experimental/LieAlgebras/src/PathModel.jl

```diff
@@ -0,0 +1,617 @@
+##### Kashiwara Crystal #####
+
+abstract type AbstractCrystal <: AbstractAlgebra.Set end
+
+abstract type AbstractCrystalElem <: AbstractAlgebra.SetElem end
+
+# AbstractCrystalElem required methods
+
+function Base.eps(b::AbstractCrystalElem)
+  error("not implemented")
+end
+
+function phi(b::AbstractCrystalElem)
+  error("not implemented")
+end
+
+@doc raw"""
+    Apply the root operator $\tilde e_i$ to `b` in place and return the result.
+"""
+function ealpha!(b::AbstractCrystalElem, i::Int)
+  error("not implemented")
+end
+
+@doc raw"""
+    Apply the root operator $\tilde f_i$ to `b` in place and return the result.
+"""
+function falpha!(b::AbstractCrystalElem, i::Int)
+  error("not implemented")
+end
+
+@doc raw"""
+    Return the weight of `b`.
+"""
+function weight(b::AbstractCrystalElem)
+  error("not implemented")
+end
+
+# AbstractCrystal optional methods
+
+@doc raw"""
+    is_upper_normal(B::AbstractCrystal) -> bool
+
+Return `true` if `B` is upper normal, and `false` if `B` is not upper normal or it is unknown.
+In the case of `false` consult the specific implemenation for details.
+"""
+function is_upper_normal(B::AbstractCrystal)
+  return true
+end
+
+@doc raw"""
+    is_lower_normal(B::AbstractCrystal) -> bool
+
+Return `true` if `B` is lower normal, and `false` if `B` is not lower normal or it is unknown.
+In the case of `false` consult the specific implemenation for details.
+"""
+function is_lower_normal(B::AbstractCrystal)
+  return true
+end
+
+@doc raw"""
+    is_normal(B::AbstractCrystal) -> bool
+
+Return `true` if `B` is normal, and `false` if `B` is not normal or it is unknown,
+whether `B` is upper normal or lower normal.
+In the case of `false` consult the specific implemenation for details.
+"""
+function is_normal(B::AbstractCrystal)
+  return is_upper_normal(B) && is_lower_normal(B)
+end
+
+# AbstractCrystalElem provided methods
+
+# ealpha
+
+@doc raw"""
+    ealpha(b::AbstractCrystalElem, i::Int) -> AbstractCrystalElem
+
+Return the result of applying the root operator $\tilde e_i$ to `b`.
+"""
+function ealpha(b::AbstractCrystalElem, i::Int)
+  return ealpha!(deepcopy(b), i)
+end
+
+@doc raw"""
+    ealpha(b::AbstractCrystalElem, i::Int, n::Int) -> AbstractCrystalElem
+
+Return the result of applying the root operator $\tilde e_i$ to `b` `n`-times.
```

Comment:
I would recommend to avoid putting formulas directly adjacent. How about this (and of course similarly in other places that do this):
```suggestion
Return the result of `n`-times applying the root operator $\tilde e_i$ to `b`.
```
(There is probably a more elegant way to phrase this, perhaps our native speakers @JohnAAbbott or @mjrodgers have a suggestion)

Reply:
I think the original is good, but without the hyphen: 
Return the result of applying the root operator to `b` `n` times.

Or alternately
Applies the root operator to `b` `n` times.
Applies the root operator `n` times to `b`.

Edit: Looks better maybe with the `b` and `n` separated.

Reply:
Possibly:  result of `n`-fold application of the root operator to `b`.
For readability we do need to keep `b` and `n` separated (by more than just a space).